### PR TITLE
Add configuration 'compressaccessible' to VariableHandler

### PR DESCRIPTION
**Pull request by AgentOak**
When `false`, try not to rename variables that are accessible from the outside, i.e. non-private class and instance properties. If `true`, the old behavior is retained.

This configuration only has an effect if 'compressvariables' is true, i.e. when VariableHandler is registered, and, like 'compressproperties', acts as a filter that prevents some of the renames. `false` by default.

When implementing this, we encounter a problem:

- It is possible to use both local variables and properties without declaring them (although this is an `E_NOTICE`/`E_DEPRECATED` in recent PHP versions)

- The Moody TokenVM is single-pass, so we have to make a decision whether or not to map a variable when we first encounter it

Therefore, we implement this configuration on a best-effort basis:

- When we encounter a local variable or a property declaration preceeded by "private" we apply the existing mapping or create a mapping

- When we encounter a property declaration that is not preceeded by "private" we mark the variable name "untouchable". Note that a property declared without a visibility is valid and public by default

- When we encounter a class property access (`->var` or `::$var`) and have a mapping, we apply the mapping

- When we encounter a class property acces but don't have a mapping yet, we have to assume the property may be public/protected (consider that dynamically created properties are public) and mark the variable name "untouchable". This creates false-negatives where we don't rename a variable that might be turn out to be private later on

The variableMapping only maps variable names instead of tracking the origin of the variable, since that is impossible in PHP. Reused variable names (but different variables) share an entry in the variableMapping. Normally, this is fine because the variables already had the same name in the original code. However, when `compressaccessible` is disabled this can create conflicts: We might first visit a local variable or a private property, create a mapping, and later see a public/protected property declaration with the same name. In such a case we *have* to rename the variable despite being non-private to make sure we do not mix up any variables and produce invalid code.

TL:DR; This solution has both false-negatives and false-positives, but since a variable name can only be marked untouchable if we don't already have a mapping, we can at least be sure that we will not generate invalid code. And most of the public/protected properties will not be renamed :)

### CODE REVIEW

**Comment by AgentOak** [comment]
This could perhaps be improved by some approaches:

- Make a `functionFetcher` tokenHandler similiar to the `classFetcher` that remembers what function we're in (or just whether or not we a are in a function). This would allow us to differentiate local `static $foo = "bar";` variables from class properties without visibility, e.g. `class Foo { static $bar = "baz"; }`. Right now all local static variables cause false-negatives and their names are marked untouchable. (Since static is enough to declare a property and no visibility modifier equals public).

- A `functionFetcher` in combination with `classFetcher` would also allow us to tell apart property declarations from local variables, so we could use different `variableMapping` tables for local variables and class properties. (If one breaks non-static<>static accesses in older PHP versions we could also split static and instance properties). This means fewer false-positives, e.g. right now a local variable that is encountered before the declaration of a property with the same name forces the property to be renamed

- Remember all T_VARIABLE/T_OBJECT_OPERATOR tokens, and JUMP around in the token array to retroactively rename variables only if no non-private declaration was found. This allows us to eliminate all false-positives (i.e. cases where public/protected properties were accidentally renamed). This would probably require a Moody VM expert to pull off.
